apps/andi/lib/andi_web/validators/dataset_schema_validator.ex

```diff
@@ -0,0 +1,52 @@
+defmodule AndiWeb.DatasetSchemaValidator do
+  @moduledoc """
+  Used to validate dataset schemas
+  """
+
+  def validate(%{"technical" => %{"sourceType" => source_type}}  = dataset) when source_type in ["ingest", "stream"] do
+    validate_schema(dataset)
+  end
+
+  def validate(_), do: []
+
+  def validate_schema(%{"technical" => %{"schema" => schema}}) when schema in [nil, []] do
+    ["schema cannot be missing or empty"]
+  end
+
+  def validate_schema(%{"technical" => %{"sourceFormat" => "text/xml", "schema" => schema}}) do
+    validate_schema_has_selectors(schema)
+  end
+
+  def validate_schema(_), do: []
+
+  defp validate_schema_has_selectors(schema) do
+    results =
+      schema
+      |> Enum.map(&build_field_validator/1)
+      |> List.flatten()
+
+    results
```

Comment:
Is "results" required here?

Reply:
Good call.  We can just return the result of the functions.